src/commands/skullboard.py

```diff
@@ -167,105 +182,343 @@ async def edit_or_send_skullboard_message(
-            )
+        # The post date is unused, may use in future if needed.
+        for post_id, user_id, channel_id, day, frequency in hof_entries[:10]:
+            # Format the entries into a readable message
+            line = f"💀 {frequency} : https://discord.com/channels/{
+                self.db.guild_id}/{channel_id}/{post_id} from <@{user_id}>"
+            msg.append(line)
 
-    @app_commands.command(name="hof", description="Get top posts (alltime)")
-    async def hof(self, interaction: Interaction):
-        try:
-            hof_entries = await self.db.get_HOF()
-            if not hof_entries:
-                await interaction.response.send_message(
-                    "Database error or empty Hall of Fame - check the logs.",
-                    ephemeral=True,
-                )
-                return
+        msg = "\n".join(msg)
+        embed = Embed(
+            title="💀   SKULLS OF THE WEEK   💀",
+            colour=LIGHT_GREY,
+            description=msg,
+        )
+
+        response = Response(embed=embed)
+        return response
+
+    @app_commands.command(name="stats", description="Get skullboard stats")
+    @app_commands.choices(
+        timeframe=[
+            app_commands.Choice(name="week", value="w"),
+            app_commands.Choice(name="month", value="m"),
+            app_commands.Choice(name="year", value="y"),
+            app_commands.Choice(name="alltime", value="a"),
+        ]
+    )
+    @interaction_handler
+    async def stats(
+        self, interaction: Interaction, timeframe: app_commands.Choice[str]
+    ) -> Response:
+        data = []
+        title = ""
+        option = ""
+        if timeframe.value == "w":
+            option = "This Week's"
+            data = await self.db.get_7_day_histogram()
+        elif timeframe.value == "m":
+            option = "This Month's"
+            data = await self.db.get_30_day_histogram()
+        elif timeframe.value == "y":
+            option = "This Year's"
+            data = await self.db.get_365_day_histogram()
+        elif timeframe.value == "a":
+            option = "All-Time"
+            data = await self.db.get_alltime_histogram()
+        else:
+            raise Exception("Invalid option")
+        title += option
+        data = [
+            (count, frequency)
+            for count, frequency in data
+            if count > 0 and frequency > 0
+        ]  # only including post/counts for more than 0 reactions and 0 posts
+
+        if not data:
+            raise Exception("Database Error")
+
+        title += " Post Distribution"
+
+        # Collecting stats
+        count = sum([y for x, y in data])  # number of posts in total
+        above_threshold = sum(
+            [y for x, y in data if x >= self.db.threshold]
+        )  # number of posts meeting or exceeding threshold
+        percentile = round(
+            100 * above_threshold / count, 1
+        )  # percentile of posts meeting or exceeding threshold
```

Comment:
Put the respective comments above the statements to prevent Black from wrapping statements like this